### PR TITLE
Feature/task 18

### PR DESCRIPTION
**Pull request by MaxSkr**

### CODE REVIEW

File: PhotoMap/NSArray+Filter.m [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by wsc-046b on 1/18/17.
+//  Copyright © 2017 wsc-046b. All rights reserved.
+//
+
+#import "NSArray+Filter.h"
+
+@implementation NSArray (Filter)
+
+-(NSMutableArray *)filter:(FilterBlock)filterBlock {
```
Comment (dmitryafanasiev) [inline PhotoMap/NSArray+Filter.m]: return types must be imutable

File: PhotoMap/NSArray+Filter.m [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    NSMutableArray *mutableArray = [NSMutableArray new];
+    for (id object in self) {
+        if (filterBlock == nil || filterBlock(object)) {
+            [mutableArray addObject:object];
+        }
+    }
+    return mutableArray;
```
Comment (dmitryafanasiev) [inline PhotoMap/NSArray+Filter.m]: use [mutableArray copy] to make it imutable

File: PhotoMap/PhotoDetailView.m [hunk 3]
```diff
@@ -95,7 +95,9 @@ +(void) showPhotoDetailDialogOnView:(UIView *) view
 
 -(void)didMoveToSuperview {
     [super didMoveToSuperview];
-    [self show:YES];
+    if (self.superview) {
```
Comment (dmitryafanasiev) [inline PhotoMap/PhotoDetailView.m]: self.superview might be always nonnull in this method

Comment (MaxSkr) [inline PhotoMap/PhotoDetailView.m]: This method is called when the view is being dismissed. From documentation: didMoveToSuperview  - Tells the view that its superview changed. So setting superview to nil triggers the method when the view is being dismissed

**Pull request merged**